### ISSUE TITLE
How to define another pictures folder in screensaver ?

### ISSUE THREAD

**Issue by Petrouchka** [mate-desktop/mate-screensaver#88]
By default, MATE screensaver proposes only the Pictures folder if you want to use aleatory pictures in the screensaver. A new feature would be to have the possibility to choose another folder (in this folder, all files that are not pictures would be ignored).

**Comment by agwells** [comment]
`xscreensaver` by itself, lets you control this with a per-user config file: `~/.xscreensaver`

```
imageDirectory: /path/to/some/directory
```

... but it looks like `mate-screensaver` doesn't respect that file. Instead, it uses your [XDG user directories](https://wiki.archlinux.org/index.php/XDG_user_directories) "PICTURES" directory. So as a workaround, you can change that by doing:

```sh
xdg-user-dirs-update --set PICTURES /path/to/your/screensaver/pics
```

The downside is that the XDG user directories aren't exclusively used by the screensaver. It's a system-wide setting that other programs may also use as a default "pictures" directory.

**Comment by superian** [comment]
The 'cosmos' screensaver gets its images from /usr/share/backgrounds/cosmos - linking that to where you want it to use may be better all round than changing the PICTURES xdg-user-directory.

I'll let someone else try it first :)

**Comment by superian** [comment]
It turns out that it will read what it thinks are subdirectories of that directory too. Result.

**Comment by iutech-france** [comment]
> It turns out that it will read what it thinks are subdirectories of that directory too. Result.

Yep, but is that a problem (serious question, from your tone I don't get if you hint that there would be unexpected consequences) ?

**Comment by superian** [comment]
No - while I was using mate-screensaver, linking where I wanted the pictures to come from - in this case a subdirectory of ~/Pictures/ that has lots of other subdirectories - to the cosmos directory worked.

I was having problems with mate-screensaver stopping working after a while, so I've switched to xcreensaver (where jwz is right, and MATE shouldn't have tried to duplicate it in the first place.)

**Comment by iutech-france** [comment]
Thanks for the precision.
Anyway opening /usr/share/applications/screensavers/personal-slideshow.desktop and adding --location=/home/iutech/Images/Belles to the end of the line Exec=/usr/lib/mate-screensaver/slideshow worked.

The problem being of course that it's hard-coded in the general configuration file rather than in a user's configuration file.
